pkg/openstack/rabbitmq.go

```diff
@@ -153,17 +155,56 @@ func reconcileRabbitMQ(
 		if err != nil {
 			return mqFailed, ctrl.Result{}, err
 		}
+		clusterNodeTLSArgs := "-proto_dist inet_tls -ssl_dist_optfile /etc/rabbitmq/inter-node-tls.config"
 		if fipsEnabled {
-			fipsModeStr := "-crypto fips_mode true"
-
-			envVars = append(envVars, corev1.EnvVar{
-				Name:  "RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS",
-				Value: fipsModeStr,
-			}, corev1.EnvVar{
-				Name:  "RABBITMQ_CTL_ERL_ARGS",
-				Value: fipsModeStr,
-			})
+			clusterNodeTLSArgs += " -crypto fips_mode true"
 		}
+
+		envVars = append(envVars, corev1.EnvVar{
+			Name:  "RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS",
+			Value: clusterNodeTLSArgs,
+		}, corev1.EnvVar{
+			Name:  "RABBITMQ_CTL_ERL_ARGS",
+			Value: clusterNodeTLSArgs,
+		})
+	}
+
+	cms := []util.Template{
+		{
+			Name:         fmt.Sprintf("%s-config-data", rabbitmq.Name),
+			Namespace:    rabbitmq.Namespace,
+			Type:         util.TemplateTypeConfig,
+			InstanceType: "rabbitmq",
+			Labels:       map[string]string{},
+			CustomData: map[string]string{
+				"inter_node_tls.config": `[
+  {server, [
+    {cacertfile,"/etc/rabbitmq-tls/ca.crt"},
+    {certfile,"/etc/rabbitmq-tls/tls.crt"},
+    {keyfile,"/etc/rabbitmq-tls/tls.key"},
+    {secure_renegotiate, true},
+    {fail_if_no_peer_cert, true},
+    {verify, verify_peer},
+    {versions, ['tlsv1.2','tlsv1.3']}
+  ]},
+  {client, [
+    {cacertfile,"/etc/rabbitmq-tls/ca.crt"},
+    {certfile,"/etc/rabbitmq-tls/tls.crt"},
+    {keyfile,"/etc/rabbitmq-tls/tls.key"},
+    {secure_renegotiate, true},
+    {verify, verify_peer},
+    {versions, ['tlsv1.2','tlsv1.3']}
+  ]}
```

Comment:
wondering if we could also move the bellow `AdvancedConfig` to a template and pass it via a volume. This would allow the `AdvancedConfig` could be used for customization? could check this in a follow up

Reply:
as discussed with @dciabrin on slack, lets leave this for now and change moving things to templates in a follow up. so +1 from my side on this